OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -946,34 +940,47 @@ void InitializeBase(Actor self)
 			{
 				initialBaseCenter = mcv.Location;
 				defenseCenter = mcv.Location;
-				QueueOrder(new Order("DeployTransform", mcv, false));
+				DeployActor(mcv, initialBaseCenter, false, false);
 			}
 			else
 				BotDebug("AI: Can't find BaseBuildUnit.");
 		}
 
 		// Find any newly constructed MCVs and deploy them at a sensible
 		// backup location.
-		void FindAndDeployBackupMcv(Actor self)
+		void FindAndDeployMcv(Actor self)
 		{
 			var mcvs = self.World.Actors.Where(a => a.Owner == Player &&
-				Info.UnitsCommonNames.Mcv.Contains(a.Info.Name));
+			Info.UnitsCommonNames.Mcv.Contains(a.Info.Name)).ToList();
 
 			foreach (var mcv in mcvs)
 			{
 				if (!mcv.IsIdle)
 					continue;
 
-				var factType = mcv.Info.TraitInfo<TransformsInfo>().IntoActor;
-				var desiredLocation = ChooseBuildLocation(factType, Info.RestrictMCVDeploymentFallbackToBase, BuildingType.Building);
```

Comment:
With this change, the `RestrictMCVDeploymentFallbackToBase` info flag becomes unused. Can we please discuss how we want MCV deployment to work before we undo #11463 completely, please? I'm kinda sick of the back-and-forth. Let's have some direction.